nifi-extension-bundles/nifi-mongodb-bundle/nifi-mongodb-services/src/main/java/org/apache/nifi/mongodb/MongoDBControllerService.java

```diff
@@ -96,25 +101,52 @@ protected MongoClient createClient(ConfigurationContext context, MongoClient exi
         }
 
         try {
-            final String uri = context.getProperty(URI).evaluateAttributeExpressions().getValue();
+            String uri = context.getProperty(URI).evaluateAttributeExpressions().getValue();
             final String user = context.getProperty(DB_USER).evaluateAttributeExpressions().getValue();
             final String passw = context.getProperty(DB_PASSWORD).evaluateAttributeExpressions().getValue();
 
             final MongoClientSettings.Builder builder = MongoClientSettings.builder();
-            final ConnectionString cs = new ConnectionString(uri);
 
-            if (user != null && passw != null) {
+            if (StringUtils.containsIgnoreCase(uri, "authmechanism") && user != null && passw != null) {
+                // we extract all specified parameters in the URI
+                final Map<String, String> result = new LinkedHashMap<>();
+                for (String param : uri.split("&")) {
+                    String[] pair = param.split("=", 2);
+                    String key = URLDecoder.decode(pair[0], StandardCharsets.UTF_8).toLowerCase();
+                    String value = pair.length > 1 ? URLDecoder.decode(pair[1], StandardCharsets.UTF_8) : "";
+                    result.put(key, value);
+                }
+
+                uri = uri.replaceAll("(?i)authmechanism=[^&]*&?", "");
```

Comment:
Instead of parsing the entire query string into a Map, it seems like it would be cleaner to define the regular expression pattern that can be used to both remove the value from the URI and find to the value of `authMechanism`. This could also be used in the conditional itself, instead of the `containsIgnoreCase()` method.